virtual_rainforest/entry_points.py

```diff
@@ -0,0 +1,67 @@
+"""The `virtual_rainforest.entry_points` module.
+
+This module defines the entry points to the virtual_rainforest package. At the moment a
+single entry point is defined `vr_run`, which simply configures and runs a virtual
+rainforest simulation based on a set of configuration files.
+"""
+
+import argparse
+import textwrap
+
+import virtual_rainforest as vr
+from virtual_rainforest.core.config import ConfigurationError
+from virtual_rainforest.core.logger import log_and_raise
+from virtual_rainforest.main import vr_run
+
+
+def _vr_run_cli() -> None:
+    """Configure and run a Virtual Rainforest simulation.
+
+    This program sets up and runs a simulation of the Virtual Rainforest model. At
+    present this is incomplete. At the moment a set of configuration files are read in
+    based on user supplied paths, these are converted to a single configuration file,
+    which is then output for further reference. This combined configuration is then used
+    to initialise a set of models.
+    """
+
+    # Check function docstring exists, as -OO flag strips docstrings I believe
+    desc = textwrap.dedent(_vr_run_cli.__doc__ or "Python in -OO mode: no docs")
```

Comment:
Where did you find out that `-O0` strips docstrings? I'm assuming you're talking about C-flags here, in which case I'd be quite surprised: `-O0` means "disable compiler optimisations", so why this would strip docstrings out I have no idea. I'd be surprised if any level of optimisation removes docstrings.

Reply:
Yeah it seemed strange to me (and I must admit to not being 100% sure on what `python -OO` is/does), but `mypy` insisted that we provide an alternative for when `_vr_run_cli.__doc__ = None`. (This)[https://github.com/python/mypy/issues/9170] `mypy` issue suggested that this was to do with ensuring support for the `-OO` flag

Reply:
Actually here's the details from `python --help`:
"
-O     : remove assert and __debug__-dependent statements; add .opt-1 before
         .pyc extension; also PYTHONOPTIMIZE=x
-OO    : do -O changes and also discard docstrings; add .opt-2 before
         .pyc extension
"
which makes sense as `-OO` explicitly discards dosctrings. Agree that removing docstrings seems like a very intense level of optimisation though

Reply:
Oh, that's weird! Thanks for the info. I guess there's no choice then!

I assumed it was something to do with gcc's optimisation flags, but apparently not.

Reply:
You can disable `mypy` for a single line by adding `# ignore: type` to the end, which is always an option.

Reply:
That's true - forgot that! The fix is easy enough here, I guess, even if it is madly obscure. We might want to start up a document on `mypy` strategy for the repo to cover what we are going to do in particular cases so that we can keep them consistent?

Reply:
That's a good idea, though I'd probably keep it as general advice rather than having formal rules about it. It would be worth mentioning that `# ignore: type` exists and that it should only be used when necessary.

Reply:
(PS - I find I mostly end up using it for bits of code where I know a value can't possibly be `None`, but `mypy` hasn't figured that out.)

Reply:
Yeah - that is very aggravating. I've just had a thing where I want `thing = None` and then a loop looks for match to set `thing  = 'str_value'`. Then I explicitly handle the `None` case and after that use the  found `str_value`  and `mypy` throws its toys. So, instead I start with `thing = ""`, which seems less clear to me.

Reply:
Yeah -- less pythonic too.

You can also do this instead:
```py
thing: Optional[str] = None
# etc.
```

then `mypy` knows it is allowed to be `None` or a string.

Reply:
I'm sorry I'm late to this conversation, but why we need this command, to start with?

Reply:
Which command, Diego? The `textwrap.dedent` or the whole `_vr_run_cli()`?

Reply:
No, the `textwrap.dedent`. And, it is running VR with `-OO` an expected use case that needs this? Just curious - I'm not complaining about it.